### PR TITLE
Add snippet for Collections()

### PR DESCRIPTION
**Pull request by samtstern**

### CODE REVIEW

**Comment by samtstern** [comment]
@rakyll could you assign the appropriate reviewer here?

File: firestore/firestore_snippets/retrieve.go [hunk 1]
```diff
@@ -129,3 +129,20 @@ func allDocs(ctx context.Context, client *firestore.Client) error {
...
+	iter := client.Collection("cities").Doc("SF").Collections(ctx)
+	for {
+		collRef, err := iter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Found collection with id: %v\n", collRef.ID)
```
Comment (broady) [inline firestore/firestore_snippets/retrieve.go]: nit: `%s` instead of `%v`

**Comment by broady** [comment]
LGTM. one nit.

**Comment by samtstern** [comment]
@broady thanks, nit addressed.

**Pull request merged**